zipline/testing/predicates.py

```diff
@@ -552,6 +552,13 @@ def assert_array_equal(result,
         if array_decimal is None else
         partial(np.testing.assert_array_almost_equal, decimal=array_decimal)
     )
+
+    is_datetime = False
+    if result.dtype.kind == 'M' and expected.dtype.kind == 'M':
+        is_datetime = True
+        result = result.astype('int64')
+        expected = expected.astype('int64')
```

Comment:
Hmm. One thing we might want to handle here is if result and expected are both datetime dtype, but have different units. I'm not sure if we'd want those to compare the same.